userspace/libsinsp/utils.cpp

```diff
@@ -595,172 +596,31 @@ bool sinsp_utils::sockinfo_to_str(sinsp_sockinfo* sinfo, scap_fd_type stype, cha
 	return true;
 }
 
-//
-// Helper function to move a directory up in a path string
-//
-void rewind_to_parent_path(char* targetbase, char** tc, const char** pc, uint32_t delta)
+std::string sinsp_utils::concatenate_paths(std::string_view path1, std::string_view path2, ssize_t max_len)
 {
-	if(*tc <= targetbase + 1)
-	{
-		(*pc) += delta;
-		return;
-	}
-
-	(*tc)--;
-
-	while(*((*tc) - 1) != '/' && (*tc) >= targetbase + 1)
-	{
-		(*tc)--;
-	}
-
-	(*pc) += delta;
-}
+    auto p1 = std::filesystem::path(path1, std::filesystem::path::format::generic_format);
+    auto p2 = std::filesystem::path(path2, std::filesystem::path::format::generic_format);
 
-//
-// Args:
-//  - target: the string where we are supposed to start copying
-//  - targetbase: the base of the path, i.e. the furthest we can go back when
-//                following parent directories
-//  - path: the path to copy
-//
-void copy_and_sanitize_path(char* target, char* targetbase, const char* path, char separator)
-{
-	char* tc = target;
-	const char* pc = path;
-	g_invalidchar ic;
+	auto path_concat = (p1 / p2).lexically_normal();
+	std::string result = path_concat;
```

Comment:
I think you might want to check this path to string conversion. Perhaps it should be explicit by calling `path_concat.generic_string()` to avoid backslashed when run in Windows. Not sure, just advice.
EDIT: Oh, perhaps the flag you used, `std::filesystem::path::format::generic_format`, does just that?

Reply:
added as it doesn't hurt

Reply:
That was indeed the intent of `std::filesystem::path::format::generic_format` in the path conversion function above. Since format is a member of the path I'd assume that a concatenation will retain the format of the input strings.

Reply:
I have double checked. `.generic_string()` was indeed necessary :) Thanks @federico-sysdig !